biscuit-quote/src/lib.rs

```diff
@@ -456,83 +456,184 @@ impl Builder {
     }
 
     fn validate(&self) -> Result<(), error::LanguageError> {
-        if self.datalog_parameters == self.macro_parameters {
+        if self.macro_parameters.is_subset(&self.datalog_parameters) {
             Ok(())
         } else {
             let unused_parameters: Vec<String> = self
                 .macro_parameters
                 .difference(&self.datalog_parameters)
                 .cloned()
                 .collect();
-            let missing_parameters: Vec<String> = self
-                .datalog_parameters
-                .difference(&self.macro_parameters)
-                .cloned()
-                .collect();
             Err(error::LanguageError::Parameters {
-                missing_parameters,
+                missing_parameters: Vec::new(),
```

Comment:
There is no more fitting error variant for just unused parameters. For a better error message, one could be added.